web/modules/custom/dpl_url_proxy/dpl_url_proxy.routing.yml

```diff
@@ -0,0 +1,20 @@
+dpl_url_proxy.configure:
+  path: 'admin/config/services/dpl-url-proxy'
+  defaults:
+    _form: '\Drupal\dpl_url_proxy\Form\ProxyUrlConfigurationForm'
+    _title: 'Configure proxy urls'
+  requirements:
+    _permission: 'administer library agency configuration'
+  options:
+    _admin_route: TRUE
+
+dpl_url_proxy.generate_url:
+  path: '/dpl-url-proxy/generate-url'
+  defaults:
+    _controller: '\Drupal\dpl_url_proxy\Controller\DplUrlProxyController::generateUrl'
```

Comment:
This is getting to a state where we are beginning to expose an API which can be consumed by others.

The current use case is very simple but when we get to search campaigns etc. things will get more complex. I suggest that we consider which paradigm we want to work with going forward (REST, JSON:API, GraphQL) so all parties use that for future work.

Reply:
Needs discussion in the architecture forum